### ISSUE TITLE
API Extractor Request: Proper Exit

### ISSUE THREAD

**Issue by bigtimebuddy** [SukantPal/pixi-build-tools#1]
Was bumping into an issue that would love if you could improve a little. There was an Error in API extractor [here](https://github.com/pixijs/pixi.js/runs/2171709081#step:11:128), but the process kept going. I would have expected this to fail out and return a non-zero exit code. The process finished like everything was okay even though the extractor failed on this package. There are times when building with `tsc` is fine, but extractor fails (namespace, bad import, etc)

**Comment by ShukantPal** [comment]
@bigtimebuddy Hmm, I did try to ensure a non-zero exit code on failure. The CI does log `@pixi/utils failed:` - which matches this line. I wonder why the exit code is 0.

https://github.com/SukantPal/pixi-build-tools/blob/812ab6c5e3138cf7e5185436621acb8c4c020c9b/packages/api-extractor-lerna-monorepo/index.js#L99

**Comment by bigtimebuddy** [comment]
Oh I see, what's going on there, because you set `process.exitCode`, you are then invaliding that error here:

https://github.com/SukantPal/pixi-build-tools/blob/812ab6c5e3138cf7e5185436621acb8c4c020c9b/packages/api-extractor-lerna-monorepo/index.js#L106

Why not `process.exit(1)`? or `process.exitCode = process.exitCode || 0;` later on?

Don't know if this is related but, but it's curious that it reports `failed` for utils (which is correct), but then show's the stack after starting `canvas-display`.
```
@pixi/utils ---------------------------------------------
Analysis will use the bundled TypeScript version 3.9.5
@pixi/utils failed:
Error: The expression contains an import() type, which is not yet supported by API Extractor:
@pixi/canvas-display ---------------------------------------------
/home/runner/work/pixi.js/pixi.js/out/packages/utils/src/index.d.ts:82:12
    at ExportAnalyzer._tryMatchImportDeclaration (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:451:19)
    at ExportAnalyzer.fetchReferencedAstEntity (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:251:45)
    at ExportAnalyzer._tryGetExportOfAstModule (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:496:34)
    at ExportAnalyzer._getExportOfAstModule (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:469:32)
    at /home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:216:60
    at Map.forEach (<anonymous>)
    at ExportAnalyzer._collectAllExportsRecursive (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:207:48)
    at ExportAnalyzer.fetchAstModuleExportInfo (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/ExportAnalyzer.js:157:18)
    at AstSymbolTable.fetchAstModuleExportInfo (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/analyzer/AstSymbolTable.js:63:37)
    at Collector.analyze (/home/runner/work/pixi.js/pixi.js/node_modules/@microsoft/api-extractor/lib/collector/Collector.js:128:57)
Analysis will use the bundled TypeScript version 3.9.5
@pixi/canvas-display: completed successfully!
```

**Comment by ShukantPal** [comment]
Closed with #2
